mailet/api/src/test/java/org/apache/mailet/AttributeValueTest.java

```diff
@@ -486,6 +535,11 @@ void nullMapShouldThrowAnException() {
                 isThrownBy(() -> AttributeValue.of((Map) null));
         }
 
+        @Test
+        void stringMapShouldThrowAnException() {
+            assertThatThrownBy(() -> AttributeValue.ofAny(ImmutableMap.of("not an", "AttributeValue")));
```

Comment:
What of you have a list of objects, the first one being an attribute value, and the second one a string?

Then the instanciation goes fine, but you got a AttributeValue that you can't toJson, no? (FindFirst before the checks)

Reply:
>> What of you have a list of objects, the first one being an attribute value, and the second one a string?

> Then you calling code is really screwed :)

Well I will not be blocking on this but I still believe user input needs to be validated.

Anyway, how large do you expect such a structure to be? Do we think iterating on an in-memory list/map will be a bottleneck here?

Also, please write the test for this, this is an interesting one actually :-)

Reply:
We also have the case that you have a `Collection<Object>` containing `AttributeValue`s, it passes instantiation, but if you mutate it after, adding a raw type, you're also screwed.